src/storage/src/source/generator.rs

```diff
@@ -256,6 +284,19 @@ fn render_simple_generator<G: Scope<Timestamp = MzOffset>>(
                     }
                 }
                 Event::Progress(Some(offset)) => {
+                    // If we've reached the requested maximum offset, cease.
+                    if offset >= max_offset {
```

Comment:
We should filter data events with this max offset too because there is nothing preventing the generator from emitting future data (for some reason)

Reply:
Great idea, thanks.